rascal-lsp/src/main/rascal/lang/rascal/tests/rename/FormalParameters.rsc

```diff
@@ -0,0 +1,147 @@
+@license{
+Copyright (c) 2018-2023, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+module lang::rascal::tests::rename::FormalParameters
+
+import lang::rascal::tests::rename::TestUtils;
+import lang::rascal::lsp::refactor::Exception;
+
+test bool outerNestedFunctionParameter() = {0, 3} == testRenameOccurrences("
+    'int f(int foo) {
+    '   int g(int foo) {
+    '       return foo;
+    '   }
+    '   return f(foo);
+    '}
+");
+
+test bool innerNestedFunctionParameter() = {1, 2} == testRenameOccurrences("
+    'int f(int foo) {
+    '   int g(int foo) {
+    '       return foo;
+    '   }
+    '   return f(foo);
+    '}
+", cursorAtOldNameOccurrence = 1);
+
+test bool publicFunctionParameter() = {0, 1} == testRenameOccurrences("", decls = "
+    'public int f(int foo) {
+    '   return foo;
+    '}
+");
+
+test bool defaultFunctionParameter() = {0, 1} == testRenameOccurrences("", decls = "
+    'int f(int foo) {
+    '   return foo;
+    '}
+");
+
+test bool privateFunctionParameter() = {0, 1} == testRenameOccurrences("", decls = "
+    'private int f(int foo) {
+    '   return foo;
+    '}
+");
+
+@expected{unsupportedRename}
```

Comment:
Why do we expect `unsupportedRename` here?

Reply:
Because at the moment, renaming of keyword parameters is not yet supported/implemented, because at an earlier stage, we decided is was tricky and not of high priority. I was just having a look at that again. I am considering implementing it before merging this PR.